### ISSUE TITLE
JSON pattern macro

### ISSUE THREAD

**Issue by fenhl** [tomjakubowski/json_macros#14]
Feature request: a macro which works like `json!`, but in a pattern-matching context. Parentheses could be used to insert Rust's regular pattern matching. Example:

```rust
if let json_pat!({"isFoo": (Json::Boolean(flag))}) = json_blob {
    // matches {"isFoo": true} and {"isFoo": false}, and binds the value of the "isFoo" key to `flag`
    if flag { frob(); }
}
```

For additional convenience, it might also be useful if identifier bindings and the `_` and `..` syntaxes were available without parentheses.

**Comment by tomjakubowski** [comment]
Yes! I've wanted this since I started writing the library. Thanks for opening the issue.

**Comment by lfairy** [comment]
Note that we don't need a separate name, since the syntax extension API tells us whether we're in an expression or pattern context already.

**Comment by fenhl** [comment]
Good point.
